pkg/sql/backfill.go

```diff
@@ -448,6 +477,30 @@ func (sc *SchemaChanger) distBackfill(
 			if err := sc.distSQLPlanner.Run(&planCtx, txn, &plan, &recv, evalCtx); err != nil {
 				return err
 			}
+
+			// Report schema change progress. We define progress at this point
+			// as the the fraction of fully-backfilled ranges of the primary index of
+			// the table being scanned. Since we may have already modified the
+			// fraction completed of our job from the 10% allocated to completing the
+			// schema change state machine or from a previous backfill attempt,
+			// we scale that fraction of ranges completed by the remaining fraction
+			// of the job's progress bar.
+			nRanges, err := sc.nRanges(ctx, txn, spans)
```

Comment:
since part of the spans have been evaluated above it would be better to update progress as soon as we compute the spans above. Note we exit above when the spans are empty.

Also we might plan to eventually get rid of the txn used here so you could define a new txn for updating the progress. Hmmm, can you move this all to getMutationToBackfill ?